src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,151 @@
+package wolox.training.controllers;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+/**
+ * @author josefranciscodelvecchio
+ * Controller for CRUD User
+ */
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    /**
+     * This method find all users
+     * @return {@link List}<{@link User}>
+     */
+    @GetMapping
+    public List<User> findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * This method find users that match with title param
+     * @param userName: Title of the User
+     * @return {@link User}
+     */
+    @GetMapping("/user-name/{userName}")
+    public User findByUserName(@PathVariable String userName) {
+        return userRepository.findByUserName(userName)
+                .orElseThrow(UserNotFoundException::new);
+    }
+
+    /**
+     * This method find user by id param
+     * @param id: Id of the user
+     * @return {@link User}
+     * @throws UserNotFoundException when user not found
+     */
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+                .orElseThrow(UserNotFoundException::new);
+    }
+
+    /**
+     * This method creates an {@link User} with the data from user param
+     * @param user {@link User}: Data to create
+     * @return created {@link User}
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    /**
+     * This method delete user by id param
+     * @param id: Id of the user
+     * @throws UserNotFoundException when user to delete not found
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
```

Comment:
Here it should return a HTTP Status 204
```suggestion
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
```